packages/api/portfolioDrafts/createPortfolioDraft.ts

```diff
@@ -23,7 +23,11 @@ export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayPr
     created_at: now,
     updated_at: now,
     status: ProvisioningStatus.NOT_STARTED,
+    name: "",
     num_portfolio_managers: 0,
+    num_applications: 0,
+    num_environments: 0,
+    num_task_orders: 0,
```

Comment:
Do we really want to store this in the database? These are computed attributes so we're going to have to make specific efforts to keep them in sync. Not suggesting that that is wrong; just curious what our design is here.

Reply:
@kylelaker-ccpo These are initial values that will be updated in the portfolio, funding, and application steps to denormalize the data for the final portfolio draft summary (similar to how `num_portfolio_managers` was there initially). I was trying to follow the convention that was already in place, and also to ensure the type interface aligns since a test failed due to the interface shape being different. Should they be removed or we take a different approach?

Reply:
The intent is to provide a light weight solution to satisfy the My Portfolios page design.
![image](https://user-images.githubusercontent.com/77642966/133121247-a1c2641c-5f60-4e21-b6d5-684106f28591.png)
@jeffsegal-ccpo spent a good bit of time considering a design that normalized some of this data into their own database items that were children of the portfolio draft.  However, in the end concluded that such a design introduced too much complexity where we have nested hierarchies.  Including these pre-computed values at the top level of the portfolio draft (and in a PortfolioDraftSummary object) will require some efforts to keep them in sync with the stored data, but will ease the burden of retrieving the data needed for the My Portfolios page.

Apart from this listing of several Portfolio Drafts, almost all of our other access patterns are directly through a specific portfolio draft where our Partition Key gives us direct access.  By contrast the access pattern required by the My Portfolio's design however requires something different.  Definitely open to alternatives.

Reply:
Thanks Jason. In general, we aren't dealing with a normalized data set and a typical strategy for DynamoDB is to make data queryable ahead of time rather than performing JOINs and doing expensive queries, so this is in line with that notion. I also want to make sure we are biased toward reads rather than writes. It just came down to wanting to avoid performing JSON parsing over a large number of Portfolios during a GET /portfolioDrafts. That said, it's not the end of the world to go that route. However, we're likely to need quite a few other denormalized fields anyway in order to be able to perform filtering and sorting on various fields from the My Portfolios screen.